### ISSUE TITLE
Breaking Changes v7

### ISSUE THREAD

**Issue by 0xpr03** [notify-rs/notify#525]
List of things to change in v7

 - [ ] (maybe) PollWatcher: change timeout between intervals to include time spend performing work
 - [x] Config: Replace deprecated `poll_interval` signature with `poll_interval_v2`
 - [ ] (maybe) unify all places of panic/error for locking & thread starts
 - [ ] remove crossbeam
 - [ ] upgrade mio to 1.0 #560
 
 todo:
 
 - [ ] add nightly test with conditional compilation check for #592

**Comment by zeroishero** [comment]
Is there more detailed information about 2 and 3?

**Comment by 0xpr03** [comment]
Not much. We introduced poll_intervall_v2 to stay semver compliant, we should change poll_intervall for v7 and thus remove v2.

And for thread start, join etc we don't handle errors the same way everywhere. Basic idea is to just decide whether to unwrap or error and do that everywhere.

**Comment by zeroishero** [comment]
So, i checked the 2nd one. It seems poll_interval is returning Duration while poll_interval_v2 is returning Option<Duration>. So, the intended change is to just change poll_interval to return Option and change calls of poll_interval_v2 back to poll_interval? That seems easy. Or Am I missing something?

**Comment by 0xpr03** [comment]
Yep that's the plan for when we actually go for v7. Until then I'd leave it as is (in main), or we'll have a harder time making patch/fix releases.

**Comment by zeroishero** [comment]
So, is there branch for v7? Will work related to this issue be accepted? I would like to do the 2nd one as it's quite easy and I will attempt 1.

**Comment by 0xpr03** [comment]
No we normally don't have that, but you can do a PR for these things anyway.

**Comment by zeroishero** [comment]
Can you explain what the first checklist is about? I want to try that.
